registration/registration_test.go

```diff
@@ -349,8 +358,51 @@ func Test_CheckCertificate(t *testing.T) {
 	t.Run("certification check enabled", func(t *testing.T) {
 		pub, private, err := ed25519.GenerateKey(nil)
 		require.NoError(t, err)
+
+		expiration := time.Now().Add(time.Hour)
+		data, err := shared.EncodeCert(&shared.Cert{Pubkey: nodeID, Expiration: &expiration})
+		require.NoError(t, err)
+
+		defaultValidCert := &shared.OpaqueCert{
+			Data:      data,
+			Signature: ed25519.Sign(private, data),
+		}
+		pubKeyHint := pub[:shared.CertPubkeyHintSize]
+
 		powVerifier := mocks.NewMockPowVerifier(gomock.NewController(t))
 
+		const keysNum = 3
+
+		dir := t.TempDir()
+		t.Cleanup(func() {
+			os.RemoveAll(dir)
+		})
```

Comment:
`t.TempDir()` already adds a `t.Cleanup` `func` that deletes the created directory after the test ends

```suggestion
		dir := t.TempDir()
```